ksqldb-engine/src/main/java/io/confluent/ksql/query/PullQueryWriteStream.java

```diff
@@ -211,6 +207,7 @@ private PullQueryRow pollRow() {
     if (monitor.enterIf(atHalfCapacity)) {
       try {
         drainHandler.forEach(h -> h.handle(null));
+        drainHandler.clear();
```

Comment:
It seems like other implementors only allow one handler at a time, and they just clobber any previous one rather than keeping a list, but I couldn't find good documentation on this.

I also think proper callers also reregister after every callback they get, but I couldn't find a very good documentation of this either.

Ultimately, the only user of these APIs are the http response objects, so if we test it out with them, we're good.